src/Microsoft.Data/DataFrame.IO.cs

```diff
@@ -127,23 +127,28 @@ private static Type MaxKind(Type a, Type b)
             using (var st = createStream())
             {
                 string line = st.ReadLine();
-                int nbline = 0;
-                while (line != null && (numRows == -1 || rowline < numRows))
+                long nbline = 0;
```

Comment:
What does `nbline` do? Can you just use `rowLine` instead? Also, it should probably have a better, more readable name.